src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -200,19 +203,43 @@ public static CSharpType GetElementType(CSharpType type)
         internal static bool IsDictionary(CSharpType type)
             => IsReadOnlyDictionary(type) || IsReadWriteDictionary(type);
 
+        internal static bool IsDictionary(CSharpType type, [MaybeNullWhen(false)] out CSharpType keyType, [MaybeNullWhen(false)] out CSharpType valueType)
+        {
+            if (IsReadOnlyDictionary(type) || IsReadWriteDictionary(type))
```

Comment:
Can we call above overload function `IsDictionary(type)` directly? in that case we will not duplicate the same logic two places which may cause maintenance effort.

Reply:
Fixed